### ISSUE TITLE
export DOCKER_HOST=tcp://localhost

### ISSUE THREAD

**Issue by kiwijunglist** [Tecnativa/docker-socket-proxy#26]
Hi

Sorry for noob question.

But what does this command do:
$ export DOCKER_HOST=tcp://localhost

What will that change on my system and how could i revert the change if required.

I'm always very reluctant to change things directly on my system and I try to keep everything in docker.

I am wondered that the command above will stop me from using docker and may interfere with the scripts that are running because of this paragraph in the wiki

```
You cannot see running containers:

$ docker container ls
Error response from daemon: <html><body><h1>403 Forbidden</h1>
Request forbidden by administrative rules.
```

Currently I can run the docker command, and I'm worried that if I use that export command I will loose access to docker.

Thanks

PS: Sorry for probably idiotic question.

**Comment by stumpylog** [comment]
I'm hardly an expert, but I think that is only for testing/showing the proxy is running and denying/allowing requests as configured.  

By default, it appears the daemon listens to a unix socket.  (See [documentation](https://docs.docker.com/engine/reference/commandline/dockerd/#daemon-socket-option).)  The client (`docker` in the normal case) uses this to communicate with the daemon.  However, the client also checks for an environment variable called `DOCKER_HOST` and will instead use that address to talk to the daemon.  

So basically, that command points `docker` (the client) to the running proxy, instead of the running `dockerd`.   Thus new commands go through the proxy and may be denied.  

As for changing things on your system, `export`ed things are gone when [the shell is closed](https://stackoverflow.com/questions/7411455/what-does-export-do-in-shell-programming).  So unless you place that in your profile or something, it won't last.  Or you can `unset` it to be extra sure.
